client/shipping-label/state/selectors/errors.js

```diff
@@ -58,7 +58,22 @@ const getPackagesErrors = ( values ) => _.mapValues( values, ( pckg ) => {
 	return errors;
 } );
 
-const getRatesErrors = ( values ) => _.mapValues( values, ( ( rate ) => rate ? null : __( 'Please choose a rate' ) ) );
+const getRatesErrors = ( { values: selectedRates, available: allRates } ) => {
+	return {
+		server: _.mapValues( allRates, ( rate ) => {
+			if ( rate.errors ) {
```

Comment:
Yes, it's intended - the form errors return value will be the same as before, and the server return value will be either an array containing all the errors, or an empty array.

I added tests so that the behaviour is clarified in 3e0276feab8f6625fbf33e42a0dc8d7b30259ca8. Also tests 🎉

Note that I also changed the errors so that we're only getting the error message, and not the entire object (related to your next comment).